Fix arithmetic operator precedence

dai-chen opened the pull request:
### Description

This PR fixed arithmetic operator precedence reported in the issue below. I tried to reorder operator in `mathOperator` rule (Reference: https://github.com/antlr/grammars-v4/blob/master/sql/mysql/Positive-Technologies/MySqlParser.g4#L2598). However, reordering operator inside `mathOperator` rule didn't work at all.

Finally, I split the `mathExpressionAtom` rule and order them by precedence (Reference: https://github.com/crate/crate/blob/master/libs/sql-parser/src/main/antlr/SqlBaseParser.g4#L299).
 
### Issues Resolved

https://github.com/opensearch-project/sql/issues/293
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Yury-Fridlyand reviewed (changes requested):
Please fix this for PPL too:
```
opensearchsql> source=people | eval test = 1 + 2 * 3 | fields `test`;
fetched rows / total rows = 1/1
+--------+
| test   |
|--------|
| 9      |
+--------+
```

dai-chen commented:
> Please fix this for PPL too:
> 
> ```
> opensearchsql> source=people | eval test = 1 + 2 * 3 | fields `test`;
> fetched rows / total rows = 1/1
> +--------+
> | test   |
> |--------|
> | 9      |
> +--------+
> ```

I missed it. Let me check.

dai-chen commented:
> > Please fix this for PPL too:
> > ```
> > opensearchsql> source=people | eval test = 1 + 2 * 3 | fields `test`;
> > fetched rows / total rows = 1/1
> > +--------+
> > | test   |
> > |--------|
> > | 9      |
> > +--------+
> > ```
> 
> I missed it. Let me check.

Fixed in https://github.com/opensearch-project/sql/pull/1172/commits/bac6f3a7c7cdae0418dcb45451d5fddaf2e521a0. I found that PPL expression doesn't support parentheses. I'm not sure if this is by design for other expressions like logical and comparison. I fixed this for arithmetic expression which is required by this precedence fix. Thanks!
